### PR TITLE
feat(rust,python): add `u8`/`i8`/`u16`/`i16` parsers to CSV reader

### LINKED ISSUE #14226: `read_csv` cannot read 8 or 16-bit ints
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl

pl.DataFrame({"a": [1, 2, 3]}).write_csv("test.csv")

for dtype in [pl.UInt8, pl.Int8, pl.UInt16, pl.Int16, pl.UInt32, pl.Int32, pl.UInt64, pl.Int64]:
    try:
        # this works if the dtypes parameter is used instead of schema
        df = pl.read_csv("test.csv", schema={"a": dtype})
    except Exception:
        print(f"{dtype} failed")
```

### Log output

```shell
UInt8 failed
Int8 failed
UInt16 failed
Int16 failed
```

General error:

```
polars.exceptions.ComputeError: unsupported data type when reading CSV: u8 when reading CSV
```

### Issue description

`UInt8`/`Int8`/`UInt16`/`Int16` cannot be provided to the schema in `read_csv` or an error occurs.

### Expected behavior

Should read in its as provided dtypes.

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.6
Index type:           UInt32
Platform:             Windows-10-10.0.19045-SP0
Python:               3.11.7 (tags/v3.11.7:fa7a6f2, Dec  4 2023, 19:24:49) [MSC v.1937 64 bit (AMD64)]

----Optional dependencies----
adbc_driver_manager:  <not installed>
cloudpickle:          <not installed>
connectorx:           0.3.2
deltalake:            <not installed>
fsspec:               <not installed>
gevent:               <not installed>
hvplot:               <not installed>
matplotlib:           3.8.2
numpy:                1.26.2
openpyxl:             3.1.2
pandas:               2.1.4
pyarrow:              14.0.1
pydantic:             <not installed>
pyiceberg:            <not installed>
pyxlsb:               <not installed>
sqlalchemy:           2.0.23
xlsx2csv:             0.8.1
xlsxwriter:           3.1.9
```

</details>

### PR DESCRIPTION
**Pull request by mcrumiller**
Resolves #14226.

### CODE REVIEW

**Comment by ritchie46** [comment]
> @ritchie46 I'm not sure why the parsers weren't implemented for these dtypes--it didn't require anything special, just boilerplate.

Because it needs to compile more code -> leads to larger binaries.

**Comment by mcrumiller** [comment]
> > @ritchie46 I'm not sure why the parsers weren't implemented for these dtypes--it didn't require anything special, just boilerplate.
> 
> Because it needs to compile more code -> leads to larger binaries.

Do we want to leave this out then? CSV parsing is used pretty often, seems like it would be good to support the smaller dtypes directly.

**Comment by ritchie46** [comment]
Yes, I am inclined to allow it for those dtypes (feature flagged). However you need to removes those `dtypes` then in the `to_cast` function

**Comment by mcrumiller** [comment]
Thanks. I've marked this as draft since it's obviously not ready yet.

File: crates/polars-io/src/csv/buffer.rs [hunk 1]
```diff
@@ -671,6 +767,14 @@ impl Buffer {
                 missing_is_null,
                 None,
             ),
+            UInt64(buf) => <PrimitiveChunkedBuilder<UInt64Type> as ParsedBuffer>::parse_bytes(
```
Comment (mcrumiller) [inline crates/polars-io/src/csv/buffer.rs:770]: I just swapped the u32/u64 order to put u64 after, to be consistent.

**Comment by mcrumiller** [comment]
@ritchie46 all ready I think.

**Comment by ritchie46** [comment]
@mcrumiller can you still remove these `dtypes` from `to_cast`?

**Comment by mcrumiller** [comment]
Didn't I do there [here](https://github.com/pola-rs/polars/pull/14241/files#diff-303864c4f16fda174d90b7b6ce53b7772f0c94e0f4937302f011c23189d5df51L448-L453), or is there somewhere else I need to as well?

**Comment by ritchie46** [comment]
> @ritchie46 didn't I do that [here](https://github.com/pola-rs/polars/pull/14241/files#diff-303864c4f16fda174d90b7b6ce53b7772f0c94e0f4937302f011c23189d5df51L448-L453), or is there somewhere else I need to as well?

Yeap, that's the one. Thanks

**Review by ritchie46 — APPROVED** [review]

**Pull request merged**